### PR TITLE
feat: 🎸 update types to 6.0

### PR DESCRIPTION
**Pull request by polymath-eric**
Converted to draft to address changes in: https://github.com/PolymeshAssociation/Polymesh/pull/1493

### Description

Adds support for 6.0 chains. The user facing API remains unchanged, but internally there were changes in terms of extrinsic  used on chain. Effort had been made to preserve the public API as is, leaving v6.0 feature updates for later PRs. 

e.g. `InstructionStatus` can now be differentiated between `Rejected` and `Success`, but the exposed status remains `Executed` to provide a least disruptive update. The more detailed version will be updated in a future breaking change.

The update mostly deals with the chain API changes. Notable areas of work were:
- Instruction legs can now be fungible, NFT or off chain. Only fungible is currently supported
- Portfolio movements can now be fungible, NFT or off chain. Only fungible is currently supported
- Instruction status is no longer pruned from chain storage. Currently "Success" and "Rejected" and presented as the historical "Executed". They will be differentiated in future releases.
- Modify signer permissions uses an updated extrinsic. The public API remains the same.
- The chain received new batch extrinsics, but migrated previous ones with "old" suffix. Currently using "old" to preserve previous behavior.

### Breaking Changes

- drops 5.x support for chains. Connected node must be v6.0  **Note**: The public API should be non-breaking from a code perspective, but this SDK version **requires** a v6.0 chain
- `Identity.getVenues` no longer supports subscription
- `InstructionStatus.Executed` is deprecated. InstructionStatus now distinguished between `Success` and `Rejected`
- Investor Uniqueness methods have been removed 
- `Instruction.reschedule` has been removed. Failed instructions can use `executeManually` instead now **(TODO: update guard in executeManually)**

### JIRA Link

[DA-819](https://polymesh.atlassian.net/browse/DA-819)

### Checklist

- [ ] Updated the Readme.md (if required) ?

[DA-819]: https://polymesh.atlassian.net/browse/DA-819?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/api/entities/Instruction/index.ts [hunk 1]
```diff
@@ -613,4 +634,23 @@ export class Instruction extends Entity<UniqueIdentifiers, string> {
...
+    switch (status) {
+      case InternalInstructionStatus.Pending:
+        return InstructionStatus.Pending;
+      case InternalInstructionStatus.Failed:
+        return InstructionStatus.Failed;
+      case InternalInstructionStatus.Rejected:
+        // return executed for backwards compatibility. Should be updated to be explicit
+        return InstructionStatus.Executed;
+      case InternalInstructionStatus.Success:
+        return InstructionStatus.Executed;
```
[the remark this answers is not in the document]
Comment (polymath-eric) [reply src/api/entities/Instruction/index.ts:651]: Makes sense. I will update it

File: src/api/entities/Instruction/types.ts [hunk 2]
```diff
@@ -5,8 +5,13 @@ import { EventIdentifier } from '~/types';
...
 export enum InstructionStatus {
   Pending = 'Pending',
-  Executed = 'Executed',
   Failed = 'Failed',
+  Rejected = 'Rejected',
+  Success = 'Success',
+  /**
+   * @deprecated v6.0 chain will specify Success or Rejected. "Executed" is kept for historical instructions
+   */
+  Executed = 'Executed',
```
Comment (polymath-eric) [inline src/api/entities/Instruction/types.ts:14]: @prashantasdeveloper I kept this for when interacting with middleware. Not sure if we could completely remove it or not though

Comment (prashantasdeveloper) [reply src/api/entities/Instruction/types.ts:14]: We can remove this. Executed in Middleware should point as Success in SDK

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**